### DISCUSSION TITLE
Accept HTTP methods other than only GET for REST requests

### DISCUSSION THREAD

**Discussion by fedebona** [Dedac/RestDataMappingPicklist#44] [Ideas]
It would be great to accept a configuration parameter for the method to be used for REST call
![clip](https://user-images.githubusercontent.com/13332721/134872398-89560d0b-cba1-4a32-a311-0296001d5add.jpg)

**Comment by Dedac** [comment]
Generally the goal is to get an array of items to map with the drop down - this kind of information is widely a  GET call, but I assume you are asking because you have a case where the API you are interested in is returning data only from a POST.  HEAD shouldn't be returning any data that would be useful. 

In general, I don't like the idea of adding POST as an option because it is a call that would modify data instead of return data.  We wouldn't want to modify data every time a work item was opened.

GET usually allows query parameters that we are already accounting for in the URL. 
POST would need to be additional configuration around the body of the message being sent.

**Reply by fedebona** [reply]
It makes absolutely sense. Sometimes REST services are not well architected and you can see read services working with POST against every best practice, but it's better to go ahead with your approach and accept only GET

**Reply by spottedmahn** [reply]
>Sometimes REST services are not well architected

unfortunate but true

**Reply by Dedac** [reply]
Thanks for your feedback @fedebona I do appreciate the ideas for improvement here.
